XenModel/XenAPI-Extensions/VM.cs

```diff
@@ -756,9 +766,13 @@ public VirtualisationStatus GetVirtualisationStatus
                     if (vm_guest_metrics != null && vm_guest_metrics.PV_drivers_installed)
                     {
                         if (vm_guest_metrics.PV_drivers_up_to_date)
-                            return VirtualisationStatus.OPTIMIZED;
+                        {
+                            return HasNewVirtualisationStates ? GetNewVirtualisationFlags() : VirtualisationStatus.IO_DRIVERS_INSTALLED | VirtualisationStatus.MANAGEMENT_INSTALLED;

```

Comment:
We seem to call GetNewVirtualisationFlags() in each state, sometimes logical-ORed with the other flags, and sometimes on its own. It would probably be neater to extract
if (HasNewVirtualisationStates)
   return GetNewVirtualisationFlags();
higher up.